doc/api/packages.md

```diff
@@ -83,6 +85,41 @@ future-proof the package in case the default type of Node.js ever changes, and
 it will also make things easier for build tools and loaders to determine how the
 files in the package should be interpreted.
 
+### Modules loaders
+
+Node.js has two system for resolving a specifier and load modules.
+
+There is the CommonJS module loader:
+
+* It is fully synchronous.
+* It is monkey patchable.
+* When resolving a specifier, if no exact match is found, it will try to add
+  extensions (`.js`, `.json`, and finally `.node`).
```

Comment:
```suggestion
* When resolving a specifier, if no exact match is found, it will try to add
  extensions (`.js`, `.json`, and finally `.node`) and index files
  (`/index.js`, `/index.json`, and finally `/index.node`).
```

Reply:
I don't think that's true, if the specifier points to an existing directory, then an exact match is found – we have a [Folders as modules](https://nodejs.org/api/modules.html#folders-as-modules) section that explains it. Otherwise, we would have to specify it first tries to load the `"main"` field if the directory contains a `package.json` file, we might as well just link to that section.

Reply:
Then maybe mention the folder resolution without spelling it out?

```
* When resolving a specifier, if no exact match is found, it will try to add
  extensions (`.js`, `.json`, and finally `.node`) and then attempt to resolve
  [folders as modules][].

Reply:
Does that work? https://github.com/nodejs/node/blob/93d6a233cf3afb504b62c6de9396e9edccd1da57/doc/api/packages.md#L97-L99